app/main.py

```diff
@@ -1,8 +1,24 @@
 class Person:
-    # write your code here
-    pass
+    people: dict[str, "Person"] = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(data: list[dict[str, int | str]]) -> list[Person]:
+    Person.people = {}
+
+    persons: list[Person] = []
+    for item in data:
+        persons.append(Person(item["name"], item["age"]))
```

Comment:
This violates checklist item https://github.com/mate-academy/py-person-class/pull/3: "When creating a list of instances from a collection (such as a list of dictionaries), it is recommended to use list comprehension rather than traditional for loops." The code builds the persons list with a for loop and append. Consider using a list comprehension (for example: persons = [Person(item["name"], item["age"]) for item in data]) and then keep the second pass for spouse linking to satisfy the checklist .